qf_lib/data_providers/sp_global/sp_data_provider.py

```diff
@@ -0,0 +1,353 @@
+            PriceField.High: SPField.HighPrice,
+            PriceField.Low: SPField.LowPrice,
+            PriceField.Close: SPField.ClosePrice,
+            PriceField.Volume: SPField.Volume
+        }
+
+    def get_history(self, tickers: Union[SPTicker, Sequence[SPTicker]], fields: Union[SPField, Sequence[SPField]],
+                    start_datetime: datetime, end_datetime: datetime = datetime.now(),
+                    frequency: Frequency = Frequency.DAILY, to_usd: bool = False, **kwargs) \
+            -> Union[float, QFSeries, QFDataFrame, QFDataArray]:
+        """
+        Parameters
+        ----------
+        tickers: SPTicker, Sequence[SPTicker]
+            tickers for securities which should be retrieved
+        fields: SPField, Sequence[SPField]
+            fields from the Market Data dataset, that should be retrieved. To see available fields please refer to
+            SPDataProvider(..).supported_fields.
+        start_datetime: datetime
+            date representing the beginning of historical period from which data should be retrieved
+        end_datetime: datetime
+            date representing the end of historical period from which data should be retrieved;
+            if no end_date was provided, by default the current date will be used.
+        frequency: Frequency
+            frequency of the data. Data is fetched always on a daily basis and then if Frequency is lower than daily,
+            it is being resampled.
+        to_usd: bool
+            flag indicating whether all pricing data should be converted to USD (by default False)
+
+        """
+        if frequency != Frequency.DAILY:
+            raise NotImplementedError("SPDataProvider get_history() supports currently only daily frequency.")
+
+        tickers, got_single_ticker = convert_to_list(tickers, SPTicker)
+        tid_to_tickers = {t.tradingitem_id: t for t in tickers}
+        fields, got_single_field = convert_to_list(fields, SPField)
+        start_datetime = self._adjust_start_date(start_datetime, frequency)
+        got_single_date = self._got_single_date(start_datetime, end_datetime, frequency)
+        
+        dfs = []
+        
+        # Fetch prices
+        price_fields = [f for f in fields if f in SPField.price_fields()]
+        if price_fields:
+            _prices = self._fetch_pricing_data(tid_to_tickers.keys(), price_fields, start_datetime, end_datetime, to_usd)
+            dfs.append(_prices)
+            
+        # Fetch all remaining fields            
+        field_to_fetching_method = {
+            SPField.DivYield: self._fetch_dividend_yield(tid_to_tickers.keys(), start_datetime, end_datetime),
+        }
+
+        for field in [f for f in fields if f not in price_fields]:
+            try:
+                dfs.append(field_to_fetching_method[field])
+            except KeyError:
+                raise ValueError(f"Field {field} is not supported by the SPDataProvider.") from None
```

Comment:
I'd raise immediately at the start if any of the fields are not in supported_fields, similar to frequency check